allspice/utils/bom_generation.py

```diff
@@ -240,8 +296,20 @@ def generate_bom_for_orcad(
             if group_column not in columns:
                 raise ValueError(f"Group by column {group_column} not found in selected columns")
     components = list_components_for_orcad(allspice_client, repository, dsn_path, ref)
-    mapped_components = _map_attributes(components, columns)
-    bom = _group_entries(mapped_components, group_by)
+
+    columns_mapping = {
+        column_name: (
+            column_config
+            if isinstance(column_config, ColumnConfig)
+            else ColumnConfig(attributes=column_config)
+        )
+        for column_name, column_config in columns.items()
+    }
+
+    mapped_components = _map_attributes(components, columns_mapping)
+    bom = _group_entries(mapped_components, group_by, columns_mapping)
+    bom = _sort_columns(bom, columns_mapping)
```

Comment:
Two things

1. Should we not filter before sorting?
2. Its unclear to me that grouping before filtering makes sense. Do you have a use case? (WHERE vs HAVING in SQL)

Reply:
We should filter before sorting, I'll change that. 

Grouping before filtering IMO makes it easier to debug a bad filter. It is harder to figure out what's wrong when your filter removes one component in a group vs the entire group. For example, consider a filter that for whatever reason removes only X3 from a group that would've included X1..5. If we filter before grouping, you wouldn't notice it unless you're paying close attention. If we filter after grouping, it would remove the entire group, which is easier to notice because one of your parts is missing entirely. 

The downside of filtering after grouping is that you can't remove just X3, but I couldn't think of reasons that would happen. If that's useful then we must filter before grouping.

Reply:
I moved filtering before sorting. I'm not resistant to changing filtering to happen before grouping as well if everyone else agrees that makes more sense.